lib/mzgl/gl/api/OpenGLAPI.h

```diff
@@ -0,0 +1,337 @@
+#pragma once
+
+#include "GraphicsAPI.h"
+#include "mzOpenGL.h"
+
+class OpenGLAPI : public GraphicsAPI {
+public:
+	OpenGLAPI(Graphics &g)
+		: GraphicsAPI(g) {}
+
+	void init() override {
+		glGetIntegerv(GL_FRAMEBUFFER_BINDING, &g.defaultFBO);
+
+		const unsigned char *openglVersion = glGetString(GL_VERSION);
+		if (openglVersion != nullptr) {
+			Log::d() << "OpenGL Version: " << openglVersion;
+#ifdef MZGL_GL2
+			Log::d() << "MZGL Compiled with MZGL_GL2==TRUE";
+#else
+			Log::d() << "MZGL Compiled with MZGL_GL2==FALSE";
+#endif
+		} else {
+			Log::d() << "OpenGL Version: null";
+		}
+		loadDefaultShaders();
```

Comment:
Could all this code go in a .cpp file? It makes the interfaces much easier to read and understand